docs/language-usage.rst

```diff
@@ -0,0 +1,354 @@
+Using the API
+=============
+
+The `Google Natural Language`_ API can be used to reveal the
+structure and meaning of text via powerful machine
+learning models. You can use it to extract information about
+people, places, events and much more, mentioned in text documents,
+news articles or blog posts. You can use it to understand
+sentiment about your product on social media or parse intent from
+customer conversations happening in a call center or a messaging
+app. You can analyze text uploaded in your request or integrate
+with your document storage on Google Cloud Storage.
+
+This is a Beta release of Google Cloud Natural Language API. This
+API is not intended for real-time usage in critical applications.
+
+.. _Google Natural Language: https://cloud.google.com/natural-language/docs/getting-started
+
+Client
+------
+
+:class:`~gcloud.language.client.Client` objects provide a
+means to configure your application. Each instance holds
+both a ``project`` and an authenticated connection to the
+Natural Language service.
+
+For an overview of authentication in ``gcloud-python``, see
+:doc:`gcloud-auth`.
+
+Assuming your environment is set up as described in that document,
+create an instance of :class:`~gcloud.language.client.Client`.
+
+  .. code-block:: python
+
+     >>> from gcloud import language
+     >>> client = language.Client()
+
+Methods
+-------
+
+The Google Natural Language API has three supported methods
+
+- `analyzeEntities`_
+- `analyzeSentiment`_
+- `annotateText`_
+
+and each method uses a `Document`_ for representing text. To
+create a :class:`~gcloud.language.document.Document`,
+
+  .. code-block:: python
+
+     >>> text_content = (
+     ...     'Google, headquartered in Mountain View, unveiled the '
+     ...     'new Android phone at the Consumer Electronic Show.  '
+     ...     'Sundar Pichai said in his keynote that users love '
+     ...     'their new Android phones.')
+     >>> document = client.document(text_content)
+
+By default, the document's language and type are English and
+plain text:
+
+  .. code-block:: python
+
+     >>> document.language
+     'en'
+     >>> document.doc_type
+     'PLAIN_TEXT'
+
+These values can also be specified in the constructor
+
+  .. code-block:: python
+
+     >>> html_content = """\
+     ... <html>
+     ...   <head>
+     ...     <title>Story Time</time>
+     ...   </head>
+     ...   <body>
+     ...     <p>La vaca salt&oacute; sobre la luna.</p>
+     ...   </body>
+     ... </html>
+     ... """
+     >>> document = client.document(html_content,
+     ...                            language='es',
+     ...                            doc_type=language.Document.HTML)
+
+The ``language`` arugment can be either ISO-639-1 or BCP-47 language
+codes; at the time, only English, Spanish, and Japanese are supported.
+However, the ``analyzeSentiment`` method only supports English text.
```

Comment:
Can we link off somewhere else for these validations? If the API adds support for new ones, we'll have to maintain this, which sucks. If we say "the list of supported languages is here" then we're in decent shape and the common case (English here), we're covered as people can assume it's there from examples.